ext/net/01_net.js

```diff
@@ -196,19 +196,25 @@ class Conn {
 
 class TcpConn extends Conn {
   #rid = 0;
+  // whether the connectino is backed by a
+  // full TCP stream (as opposed to an UpgradeStream)
+  #isTcpStream = true;
 
-  constructor(rid, remoteAddr, localAddr) {
+  constructor(rid, remoteAddr, localAddr, isTcpStream) {
```

Comment:
I guess this is fine, but it's super confusing, the `TcpConn` class checks if it's TCP stream. Should we consider adding more subclasses to better discriminate between various kinds?

Reply:
Huh what other streams is `TcpConn` used for?

Reply:
Yeah you're right that's cleaner, done

Reply:
It's also used for `UpgradeStream`s